Tests/SecureXPCTests/Encoder & Decoder/Round Trip/File Descriptor XPC Wrapper Tests.swift

```diff
@@ -0,0 +1,194 @@
+//
+//  FileDescriptorXPCContainer Tests.swift
+//  SecureXPC
+//
+//  Created by Josh Kaplan on 2022-07-10
+//
+
+import System
+import Foundation
+import XCTest
+import SecureXPC
+
+final class FileDescriptorXPCContainerTests: XCTestCase {
+    // MARK: helper functions
+    
+    private func currentPath(filePath: String = #filePath) -> String { filePath }
+    
+    private func pathForFileDescriptor(fileDescriptor: Int32) -> String {
+        var buffer = [CChar](repeating: 0, count: Int(PATH_MAX))
+        _ = fcntl(fileDescriptor, F_GETPATH, &buffer)
+        
+        return String(cString: buffer)
```

Comment:
Wow I'm baffled to learn that [`Foundation.FileHandle`](https://developer.apple.com/documentation/foundation/filehandle) doesn't have an API for this.

I was hoping to avoid dealing with overly terse C gobbledygook, but looks like that's the only option here!

Reply:
Yeah, I was expecting `System.FileDescriptor` to have an API for this since it seems like it's meant to be a nice object oriented wrapper over C functionality.